Add social panel

xualexander opened the pull request:
## Types of changes
<!--- What types of changes does your code introduce to UCLA Radio? Put an `x` in the boxes that apply. -->

- [ ] Bugfix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Refactor (change which changes the codebase without affecting its external behavior)
- [ ] Non-breaking change (fix or feature that would causes existing functionality to work as expected)
- [ ] Breaking change (fix or feature that would cause existing functionality to __not__ work as expected)

## Purpose
Adds a social media link panel to the home page.
Resolves #83 
## Approach
Wrote a React component that generates Font Awesome icons as links to various social media pages, and added it to the FrontpageContent component. 
Uses an array of objects that stores the base icon name and URL for each link, in an effort to be more maintainable.
## Learning
<!--- Describe the research stage. Link to any blog posts, video, patterns, libraries, addons, or other resources that helped you to solve this problem. -->

## Screenshot(s) 
<img width="290" alt="screen shot 2017-11-03 at 17 21 31" src="https://user-images.githubusercontent.com/19811816/32400344-1f5d0d56-c0bc-11e7-8e10-83fda19f5250.png">

## Checklist
<!--- Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code. -->

- [x] My branch follows the branch naming scheme of UCLA Radio, and can merge into `master` without error.
- [x] My code follows the code style of this project, and I have linted it to confirm this.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

## Platform
<!--- (if applicable--you can delete otherwise) -->
__Operating System:__ macOS Sierra
__Node Version:__ 6.11.5

## Further Comments
Tested on my dev environment!

nathunsmitty commented on client/react/frontpage/components/SocialPanel.js:
```diff
@@ -0,0 +1,51 @@
+    return 'fa fa-' + img + ' fa-' + size;
+  },
+  render() {
+    const self = this;
```
Why did you declare `self` here vs just using `this.convertToFaName` in line 38?

xualexander replied:
Oops, was just confused about binding in React. Fixed.

nathunsmitty commented on client/react/frontpage/components/SocialPanel.js:
```diff
@@ -0,0 +1,51 @@
+    const self = this;
+    return (
+      <div>
+        <p className="infoHeader">Social Media</p>
```
Hmm I'm not a big fan of the header saying "Social Media"—I feel like it's a little redundant. However, not having the banner also looks weird...

Lemme dig up some old mockups and see if I can suggest a better designed solution.

xualexander replied:
Removed the header for now

nathunsmitty commented on client/react/frontpage/components/SocialPanel.js:
```diff
@@ -0,0 +1,51 @@
+            // change size here
+            let faName = self.convertToFaName(obj.img, '2x');
+            return (
+              <a href={obj.link} className="social-link">
```
You should check out the code in [ShowPage.js](https://github.com/uclaradio/uclaradio/blob/master/client/react/frontpage/components/ShowPage.js), [ShowPage.scss](https://github.com/uclaradio/uclaradio/blob/master/client/react/frontpage/components/ShowPage.scss), and [_common.scss](https://github.com/uclaradio/uclaradio/blob/master/client/react/frontpage/_common.scss). I implemented a similar feature for shows, and I think you could steal some styles from that.

Also, I really like how you implemented this with `.map`, so if you want to refactor ShowPage.js, that'd be 💯

binerys replied:
+1 to your use of map to produce this social media panel. I'd love to see your Social Media panel utilized in our ShowPage component.

binerys commented on client/react/frontpage/components/SocialPanel.js:
```diff
@@ -0,0 +1,51 @@
+  },
+];
+
+const SocialPanel = React.createClass({
```
Let's convert this component into a [stateless functional component](https://medium.com/@joshblack/stateless-components-in-react-0-14-f9798f8b992d). I linked to a medium post that describes what it is and an example of how to do it. Essentially, our component is not holding any state and simply needs to return markup. 

If you end up following the guide I linked, don't worry about the part where he mentions `propTypes`. We don't have that yet. Let me know if you have any questions (especially with all the ES6 arrow function stuff).

xualexander replied:
Refactored!

xualexander commented:
I refactored SocialPanel and updated Frontpage.js to show how we could reuse the component for ShowPage and ShowBlurb. However, the social media links for each show are currently stored in an object that requires checking whether the social media exists whenever social media icons are rendered.

If we change the social media object(s) for each show to be an array of objects (as shown in Frontpage.js), it will be more maintainable overall (if social media sites/icons change) and allow for cleaner reuse of SocialPanel. However, this would involve changing the Panel API. I was curious as to what you guys think, and how we should go about refactoring the API if this change is worthwhile.

nathunsmitty commented on client/react/frontpage/components/SocialPanel.js:
```diff
@@ -0,0 +1,18 @@
+const SocialPanel = ({ links, size }) => (
+  <div className="link-container">
+    {links.map(function(obj) {
+      let faName = 'fa fa-' + obj.img + ' fa-' + size;
```
Prefer [string template literals](https://github.com/airbnb/javascript#es6-template-literals) over concatenation.

nathunsmitty commented on client/react/frontpage/components/SocialPanel.js:
```diff
@@ -0,0 +1,18 @@
+
+const SocialPanel = ({ links, size }) => (
+  <div className="link-container">
+    {links.map(function(obj) {
```
I'd use an [arrow function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) here.

haejinjo commented:
I noticed there's no interaction when you hover over the buttons. It would be cool if they could change color or highlight or something for the *~*UX*~**~~

binerys commented:
moving forward with merging this, great job! 👍
